### PR TITLE
[FLINK-23401][python] Separate data and timer processing into different channels

### PR DESCRIPTION
**Pull request by dianfu**
## What is the purpose of the change

*This pull request separate data and timer processing into different channels.*

## Verifying this change

This change is already covered by existing tests, such as test_keyed_co_process_function*.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable)

### CODE REVIEW

File: flink-python/pyflink/fn_execution/coder_impl_slow.py [hunk 1]
```diff
@@ -114,7 +114,7 @@ class ValueCoderImpl(LengthPrefixBaseCoderImpl):
     def __init__(self, field_coder: 'FieldCoderImpl'):
         super(ValueCoderImpl, self).__init__(field_coder)
 
-    def encode_to_stream(self, value, out_stream):
+    def encode_to_stream(self, value, out_stream: OutputStream):
```
Comment (HuangXingBo) [inline flink-python/pyflink/fn_execution/coder_impl_slow.py:117]: What about add the type hint to all `encode_to_stream` and `decode_from_stream`?

File: flink-python/pyflink/fn_execution/datastream/timerservice_impl.py [hunk 2]
```diff
@@ -18,49 +18,30 @@
...
+from io import BytesIO
 
+from apache_beam.runners.worker.bundle_processor import TimerInfo
+from apache_beam.transforms import userstate
+from apache_beam.transforms.window import GlobalWindow
+
+from pyflink.common import Row
 from pyflink.datastream import TimerService
-from pyflink.fn_execution.datastream.timerservice import InternalTimer, K, N, InternalTimerService
+from pyflink.fn_execution.datastream.timerservice import N, InternalTimerService, InternalTimer, K
```
Comment (HuangXingBo) [inline flink-python/pyflink/fn_execution/datastream/timerservice_impl.py:29]: unnecessary change?

**Review by HuangXingBo — COMMENTED** [review]
@dianfu Thanks a lot for the PR. Looks good overall. I have left some comments

**Comment by dianfu** [comment]
@HuangXingBo Thanks a lot for the review. Have updated the PR accordingly.

**Review by HuangXingBo — APPROVED** [review]
Thanks a lot for the update. LGTM
